src/components/Image/Image.tsx

```diff
@@ -84,14 +97,27 @@ export class Image extends React.Component<ImageProps> {
       width,
       height,
       onLoad,
+      aspectRatio,
       resize,
       loadingBehavior,
       ...imageProps
     } = this.props;
     const { isLoaded } = this.state;
-    const dimensions = { width, height };
 
     const isAbsoluteUrl = src && src.match('^https?://');
+
+    const aspectRatioAsNumber =
+      typeof aspectRatio === 'number'
+        ? aspectRatio
+        : AspectRatioOptions[aspectRatio];
+    const sourceDimensions = { width, height };
+    const containerDimensions = {
+      width:
+        !width && aspectRatioAsNumber ? height * aspectRatioAsNumber : width,
+      height:
+        !height && aspectRatioAsNumber ? width / aspectRatioAsNumber : height,
+    };
```

Comment:
We might improve this logic a little later

Reply:
Is the case that both width and height are not defined is possible?

Reply:
Currently we don't support it since MediaImage of ui-core must get width & height... But we're planning to support this use case by inferring the values from the parent in DOM